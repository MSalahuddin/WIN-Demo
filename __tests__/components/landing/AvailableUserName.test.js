

import React from 'react';
import { render } from 'react-native-testing-library';
import AvailableUserName from '../../../js/components/landing/AvailableUserName';

describe('Component ChooseUserName', () => {

    const data = ['name1','name2','name3']
    const onSelect = jest.fn();

    test('renders correctly', () => {
        const output = render(
          <AvailableUserName
            data={data}
            onSelect={onSelect}
          />
        );
        expect(output).toMatchSnapshot();
      });
});