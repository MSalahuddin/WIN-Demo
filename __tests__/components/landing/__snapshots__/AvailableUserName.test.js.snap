// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ChooseUserName renders correctly 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": 52.6,
          "justifyContent": "center",
          "width": 100,
        },
      ]
    }
  >
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#353535"
      fontFamily="REGULAR"
      isUnderlined={false}
      isUppercase={false}
      maxFontSizeMultiplier={1.3}
      size="SMALL"
      style={
        Array [
          Object {
            "color": "#353535",
            "fontFamily": "CalibreRegular",
            "fontSize": 27,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#353535",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
        ]
      }
    >
      Available:
    </Text>
  </View>,
  <RCTScrollView
    data={
      Array [
        "name1",
        "name2",
        "name3",
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 50,
              "height": 66.6,
              "justifyContent": "center",
              "marginHorizontal": 5,
              "opacity": 1,
              "width": 100,
            }
          }
        >
          <Text
            alignCenter={false}
            allowFontScaling={true}
            color="#FFFFFF"
            fontFamily="MEDIUM"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            size="SMALL"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Calibre-Medium",
                  "fontSize": 27,
                  "justifyContent": "center",
                  "textAlign": "left",
                  "textDecorationColor": "#FFFFFF",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
              ]
            }
          >
            name1
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 50,
              "height": 66.6,
              "justifyContent": "center",
              "marginHorizontal": 5,
              "opacity": 1,
              "width": 100,
            }
          }
        >
          <Text
            alignCenter={false}
            allowFontScaling={true}
            color="#FFFFFF"
            fontFamily="MEDIUM"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            size="SMALL"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Calibre-Medium",
                  "fontSize": 27,
                  "justifyContent": "center",
                  "textAlign": "left",
                  "textDecorationColor": "#FFFFFF",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
              ]
            }
          >
            name2
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 50,
              "height": 66.6,
              "justifyContent": "center",
              "marginHorizontal": 5,
              "opacity": 1,
              "width": 100,
            }
          }
        >
          <Text
            alignCenter={false}
            allowFontScaling={true}
            color="#FFFFFF"
            fontFamily="MEDIUM"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            size="SMALL"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Calibre-Medium",
                  "fontSize": 27,
                  "justifyContent": "center",
                  "textAlign": "left",
                  "textDecorationColor": "#FFFFFF",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
              ]
            }
          >
            name3
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>,
]
`;
